launcher/src/main/java/org/wso2/carbon/launcher/extensions/DropinsBundleDeployer.java

```diff
@@ -17,296 +17,46 @@
+                            logger.log(Level.SEVERE, "Failed to update the Carbon Profile : " + carbonProfile, e);
                         }
-                    } else {
-                        //  Version property is different. Therefore this new bundle does not exist in the system.
-                        //  Therefore 'found' is still false.
-                        found = false;
-                    }
+                    });
                 }
-
-                if (!found) {
-                    //  Dropins bundle is not available in the system. Hence, add it.
-                    bundleInfoLineList.add(newBundleInfoLine);
-                    logger.log(Level.FINE, "Deploying bundle: ", newBundleInfoLine.getBundlePath());
-                }
-            }
-        });
-    }
-
-    /**
-     * Updates the bundles.info file with the new OSGi bundle information, if exists.
-     *
-     * @param bundlesInfoFile   the {@link Path} instance to the existing bundles.info file
-     * @param bundleInfoLineMap the {@link Map} containing the complete OSGi bundle information
-     * @throws Exception if an error occurs when updating the existing bundles.info file
-     */
-    private static void updateBundlesInfoFile(Path bundlesInfoFile, Map<String, List<BundleInfo>> bundleInfoLineMap)
-            throws Exception {
-        //  Generates the new bundles.info file into a temp location.
-        String tempDirectory = System.getProperty("java.io.tmpdir");
-        if (tempDirectory == null || tempDirectory.length() == 0) {
-            throw new Exception("java.io.tmpdir property is null. Cannot proceed.");
-        }
-
-        Path tempBundlesInfoDirectory = Paths.get(tempDirectory, "bundles_info_" + UUID.randomUUID().toString());
-        Path tempBundlesInfoFilePath = Paths.get(tempBundlesInfoDirectory.toString(), "bundles.info");
-        if (!Files.exists(tempBundlesInfoDirectory)) {
-            Files.createDirectories(tempBundlesInfoDirectory);
-        }
-
-        if (Files.exists(tempBundlesInfoDirectory)) {
-            String[] keyArray = bundleInfoLineMap.keySet().toArray(new String[bundleInfoLineMap.keySet().size()]);
-            Arrays.sort(keyArray);
-
-            List<String> bundleInfoLines = new ArrayList<>();
-            for (String key : keyArray) {
-                List<BundleInfo> bundleInfoLineList = bundleInfoLineMap.get(key);
-                bundleInfoLineList.forEach(bundleInfoLine -> bundleInfoLines.add(bundleInfoLine.toString()));
-            }
-            Files.write(tempBundlesInfoFilePath, bundleInfoLines, Charset.forName("UTF-8"));
-
-            if (Files.exists(bundlesInfoFile)) {
-                //  Replaces the original one with the new temporary file.
-                Files.copy(tempBundlesInfoFilePath, bundlesInfoFile, StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                logger.log(Level.SEVERE, "Failed to update the specified Carbon Profile", e);
```

Comment:
If there was an IOException, shouldn't we revert everything, thrown the exception and stop the process?